apis/v1alpha2/shared_types.go

```diff
@@ -155,22 +182,59 @@ type BackendRef struct {
 // RouteConditionType is a type of condition for a route.
 type RouteConditionType string
 
+// RouteConditionReason is a reason for a route condition.
+type RouteConditionReason string
+
 const (
 	// This condition indicates whether the route has been accepted or rejected
 	// by a Gateway, and why.
-	ConditionRouteAccepted RouteConditionType = "Accepted"
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "Accepted"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	RouteConditionAccepted RouteConditionType = "Accepted"
```

Comment:
naive question, "Accepted" is a Type?

Reply:
Not sure I understand the question. We're defining it as a type along with all the corresponding reasons we recommend setting for that condition. This matches the pattern we've used elsewhere for status conditions and reasons.

Reply:
ah, I saw below reason "accepted" and type "ResolvedRefs" so I get confused, it was curiosity , disregard it